arch/X86/X86DisassemblerDecoder.c

```diff
@@ -2042,7 +2042,7 @@ static int readOperands(struct InternalInstruction *insn)
 					return -1;
 				// Apply the AVX512 compressed displacement scaling factor.
 				if (x86OperandSets[insn->spec->operands][index].encoding != ENCODING_REG && insn->eaDisplacement == EA_DISP_8)
-					insn->displacement *= 1 << (x86OperandSets[insn->spec->operands][index].encoding - ENCODING_RM);
+					insn->displacement *= (int64_t)1 << (x86OperandSets[insn->spec->operands][index].encoding - ENCODING_RM);
```

Comment:
looks like this is a fix (for warning?) unrelated to this new feature?

Reply:
from https://github.com/aquynh/capstone/commit/2d5b0a3308489896d5b08595e67db92a675a4f71 imho it should be 1ULL, ill fix it tomorrow with the other formatting

Reply:
looks like this is because `displacement` type is changed from int32 to int64. why is this necessary?

Reply:
i assumed it was as it was in the encoding branch, it very well may not be